docs/tutorials/send_receive_ingress_pem.md

```diff
@@ -0,0 +1,387 @@
+$ kubectl get issuers  -n send-receive-project -o wide send-receive-root-issuer --watch
+NAME                       READY   STATUS   AGE
+send-receive-root-issuer   True             45s
+```
+
+#### Create the issuer certificate
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: send-receive-issuer-cert
+  namespace: send-receive-project
+spec:
+  isCA: true
+  commonName: ArtemiseCloud send-receive issuer
+  dnsNames:
+    - issuer.demo.artemiscloud.io
+  secretName: send-receive-issuer-cert-secret
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+  issuerRef:
+    name: send-receive-root-issuer
+    kind: Issuer
+EOF
+```
+
+```console
+kubectl get certificates  -n send-receive-project --watch
+NAME                       READY   SECRET                            AGE
+send-receive-issuer-cert   True    send-receive-issuer-cert-secret   24s
+```
+
+
+#### Create the issuer
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: send-receive-issuer
+  namespace: send-receive-project
+spec:
+  ca:
+    secretName: send-receive-issuer-cert-secret
+EOF
+```
+
+```console
+$ kubectl get issuer  -n send-receive-project --watch
+NAME                       READY   AGE
+send-receive-issuer        True    10s
+send-receive-root-issuer   True    50s
+```
+
+### Deploy the ActiveMQ Artemis Broker
+
+#### Ingress and ssl configuration
+
+The acceptor is configured as follows:
+* `port: 62626`
+* `sslEnabled: true` activates ssl
+* `expose: true` and `exposeMode: ingress` are triggering an ingress creation
+* `sslSecret: send-receive-sslacceptor-0-svc-ing-ptls` provides the name of the
+  sercret store that'll be used to secure the connection
+* `ingressHost:
+  send-receive-ss-0.send-receive-hdls-svc.send-receive-project.svc.demo.artemiscloud.io`
+   customizes the url to access the `acceptor` from outside the cluster.
+
+An annotation is used to ask `cert-manager` to produce the `sslSecret`
+needed to secure the connection. It is configured as follows:
+* the `selector` attaches the annotation to the `acceptor`'s ingress by its name
+* `cert-manager.io/issuer: send-receive-issuer` links to the previously created
+  issuer, `cert-manager` will use it to generate the secrets.
+* `spec.hosts` list the domains for which the certificate applies, it has to
+  match  the url of the ingress set in the `acceptor`.
+* `spec.secretName` matches the `sslSecret` value of the `acceptor` and must be
+  set to the name of the ingress + `-ptls`. Here: `send-receive-sslacceptor-0-svc-ing-ptls`
+
+##### About the `ingressHost` value
+
+The `ingressHost` we're using must have the same value as the `acceptor`
+listening URL. This we are relying on the ingress DNS plugin to resolve the IP
+for the ingress.
+
+A bit of explanation about why: when the client reaches out to the ingress, it
+ends up getting the `acceptor` url to create the final connection. But the
```

Comment:
I think the crux here is that the ingress is configured to be passthrough. So it uses the SNI to route/resolve inside the cluster. The SNI is set to the HOST of the tcp:// url that our clients use. If the HOST/SNI does not match the acceptor url host, the client will be redirected to the wrong place.

There may be additional implication on how ingress_dns works, in that it has to find something to match on, but when TLS is in play, and when the acceptor is doing the TLS termination, the HOST/SNI on the client url needs to match the acceptor and its cert needs a match for host verification.

Reply:
Can I quote your comment in the documentation directly?

Reply:
sure or improve on it :-)